pdc_client/__init__.py

```diff
@@ -161,7 +258,13 @@ def decode(req):
 
         content_type = "application/json"
         encode = json.dumps
-        self.client = beanbag.BeanBag(url, session=self.session, fmt=(content_type, encode, decode))
+        # In get_paged function it returns an interator, so user can get each pages quickly in plugins.
+        # The original call function of BeanBag returns the list.
+        # Keep the current codes and options in codes. Just init MultiPageBeanBag when is_multi is true.
+        if is_multi:
+            self.client = MultiPageBeanBag(url, session=self.session, fmt=(content_type, encode, decode))
```

Comment:
> Because if we use the new class here, the worker() in get_paged function should be removed as the new class can get all the pages.

The new class gets all pages but client needs to wait until all of them are fetched regardless of the `page_size` field value (with small `page_size` you would only make more request but client still needs to wait for all pages to be fetched). I think it could be better to use `page_size=-1` so PDC server can send all results at once. But I still don't like that the `page_size` is ignored.

> And as I comment on codes, the call function return a list, when you use pdc command to list data, the list function in plugins will get an interator return from get_pages.

If it's not possible to return different type (other than list) in the the new class, is it possible to handle this in in `PDCClient.__getitem__()`?

BTW, I cannot find `BeanBag.make_request()` in [documentation](http://beanbag.readthedocs.io/en/latest/v1.html) and overriding the class methods is not mentioned at all. It could make upgrading to a new version difficult.